libs/Helpers/template/BuildConfiguration.h.in

```diff
@@ -0,0 +1,25 @@
+// Cross platform and environment build definition
+#if defined(_MSC_VER)
+    // The microsoft compiler sets _DEBUG appropriately when targeting a debug build.
+    #if _DEBUG
+        #define BUILD_DEBUG 1
+    #else
+        #define BUILD_RELEASE 1
+    #endif
+#elif defined(__apple_build_version__)
+    // Xcode might set DEBUG when targeting debug, maybe.
+    #if DEBUG
+        #define BUILD_DEBUG 1
+    #else
+        #define BUILD_RELEASE 1
+    #endif
+#else
+    #cmakedefine BUILD_DEBUG 1
+    #cmakedefine BUILD_RELEASE 1
+#endif
+
+#cmakedefine BUILD_DEVELOPMENT 1
```

Comment:
Why do we need `BUILD_DEVELOPMENT` define? Isn't it redundant to `DEBUG`? If we need some diagnosis in Release builds I opt for introducing explicit defines, like `PERFORMANCE_MEASUREMENT=true`

Reply:
I added the following previous changes. It now defaults with DEBUG but the intended purpose is to be enabled per platform and activate development-related code like hot-reloading, logs colouring that I see in the previous POC